fix: Eval helm before depending on helm-map

cashpw opened the pull request:
Fixes #29.

cashpw committed:
fix: Eval helm before depending on helm-map

Fixes #29.

renatofdds reviewed (changes requested):
Thank you for the PR!

This code is incorrect, we should not define the helm variables in this package. When you initialize a variable with `defvar` it gets bound to the location it was first defined.

Those `defvar` were added for the compiler, but since we are already requiring `helm-core` during compile-time (when it's available), they are safe to remove.

You can just extract and wrap the `set-keymap-parent` in a `(with-eval-after-load 'helm-core ...)` for now. In the future, I think is best to refactor this `helm`-related code to a separate library.

cashpw commented:
Are these changes what you had in mind?

cashpw committed:
refactor: Format with elisp-autofmt-buffer

**What?**

1. Replace tabs with spaces
1. Run `elisp-autofmt-buffer`

cashpw committed:
feat: start_at and due_at

**What?**

1. Set `SCHEDULED` and `DEADLINE` based on `due_at` and `start_at`
values when they're available. Fall back to `due_on` and `start_on`.

**Why?**

Asana's API provides both `due_at` and `due_on` and `start_*`
equivalents. `due_at` is "The UTC date and time on which this task is
due, or null if the task has no due time". Using this value, when it's
non-null, ensures the deadline/schedule time includes the time portion
and not just the date portion.

cashpw committed:
refactor: Format

cashpw committed:
feat: Add after insert task hook

**What?**

1. Add `asana-after-insert-task-hook` to run after inserting a task

**Why?**

It's a nice entry point to amend the synced tasks (for example: set a
category, effort, priority, etc).

cashpw committed:
feat: Conditionally sync task stories

**What?**

1. Only insert the task's stories (in `LOGBOOK`) when
`asana-sync-stories` is non-nil

**Why?**

I don't need these details and they clutter the file.

cashpw committed:
feat: Configure schedule vs deadline

**What?**

1. Add `asana-schedule-pattern` to allow users to determine how due
dates are represented within org-mode

**Why?**

Asana's free plan doesn't include `start_*` times.
